demos/list.html

```diff
@@ -1329,6 +1329,48 @@ <h3>Example - Interactive List (Dark Theme)</h3>
             evt.preventDefault();
           });
         }
+
+        var deactivateRippleHandler = null;
+        var checkboxListRippleSurfaceClass = 'checkbox-list-ripple-surface';
+
+        function clickHandler(evt, rippleFoundation, li) {
+          var checkbox = li.querySelector('.mdc-checkbox__native-control');
+          var isLabel = Array.prototype.some.call(checkbox.labels, function(label) { return label === evt.target; });
+          if(evt.target !== checkbox && !isLabel) {
+            checkbox.checked = !checkbox.checked;
+            li.classList.add(checkboxListRippleSurfaceClass);
+            rippleFoundation.activate();
+
+            deactivateRippleHandler = e => deactivateRipple(e, rippleFoundation, li);
+            li.addEventListener('mouseup', deactivateRippleHandler);
+          }
+        };
+
+        function deactivateRipple(evt, rippleFoundation, li) {
+          li.classList.remove(checkboxListRippleSurfaceClass);
+          rippleFoundation.deactivate();
+          li.removeEventListener('mouseup', deactivateRippleHandler);
+        };
+
+        function addRippleToListItems(list) {
+          list.querySelectorAll('.mdc-list-item').forEach(function(li) {
```

Comment:
Not all browsers support `NodeList.forEach()` (I'm looking at you, IE 11 💩 ).

So you have to do `[].forEach.call(function(li) { ... })` instead.